### PR TITLE
Stabilise server suite tests

### PR DESCRIPTION
**Pull request by nsarupr**
stabilise server suite tests

### CODE REVIEW

File: app/server/appsmith-server/src/test/java/com/appsmith/server/services/ActionCollectionServiceTest.java [hunk 1]
```diff
@@ -199,9 +202,12 @@ public void setup() {
     @AfterEach
     @WithUserDetails(value = "api_user")
     public void cleanup() {
```
Comment (trishaanand) [inline app/server/appsmith-server/src/test/java/com/appsmith/server/services/ActionCollectionServiceTest.java:204]: Clean up can be skipped no? Each user would create a new workspace. Unless we are counting workspaces/applications somewhere in the test.

File: app/server/appsmith-server/src/test/java/com/appsmith/server/services/ApplicationPageServiceTest.java [hunk 2]
```diff
@@ -52,26 +56,47 @@ public class ApplicationPageServiceTest {
...
+
+    @BeforeEach
+    public void setup() {
+        Workspace unsavedWorkspace = new Workspace();
+        unsavedWorkspace.setName("ApplicationPageServiceTest Workspace");
+        workspace = workspaceService.create(unsavedWorkspace).block();
+    }
+
+    @AfterEach
+    public void cleanup() {
```
Comment (trishaanand) [inline app/server/appsmith-server/src/test/java/com/appsmith/server/services/ApplicationPageServiceTest.java:75]: Same. If we are creating a new workspace per test, we can actually skip this right?

File: app/server/appsmith-server/src/test/java/com/appsmith/server/services/ApplicationSnapshotServiceTest.java [hunk 3]
```diff
@@ -40,21 +44,53 @@ public class ApplicationSnapshotServiceTest {
...
+
+    @Autowired
+    SessionUserService sessionUserService;
+
+    Workspace workspace;
+
+    @BeforeEach
+    public void setup() {
+        User user = sessionUserService.getCurrentUser().block();
+        if (null == user) {
```
Comment (trishaanand) [inline app/server/appsmith-server/src/test/java/com/appsmith/server/services/ApplicationSnapshotServiceTest.java:61]: Why was this required? Are there test cases which are running without a user? Do we need to do this for other files as well?

File: app/server/appsmith-server/src/test/java/com/appsmith/server/services/ce/ActionServiceCE_Test.java [hunk 4]
```diff
@@ -183,84 +195,84 @@ public class ActionServiceCE_Test {
...
-                            workspaceId, tuple.getT2(), tuple.getT1().getId(), gitData.getBranchName()))
-                    .block();
-
-            gitConnectedPage = newPageService
-                    .findPageById(gitConnectedApp.getPages().get(0).getId(), READ_PAGES, false)
-                    .block();
-
-            branchName = gitConnectedApp.getGitApplicationMetadata().getBranchName();
-        }
+        //        if (workspaceId == null) {
```
Comment (trishaanand) [inline app/server/appsmith-server/src/test/java/com/appsmith/server/services/ce/ActionServiceCE_Test.java:206]: Please remove commented code.

**Review by trishaanand — COMMENTED** [review]
Partial review. Added comments which should be valid for rest of the files as well.

File: app/server/appsmith-server/src/test/java/com/appsmith/server/services/ce/ActionServiceCE_Test.java (hunk 4 above)
Comment (nsarupr) [reply app/server/appsmith-server/src/test/java/com/appsmith/server/services/ce/ActionServiceCE_Test.java]: Removed commented code.

File: app/server/appsmith-server/src/test/java/com/appsmith/server/services/ActionCollectionServiceTest.java (hunk 1 above)
Comment (nsarupr) [reply app/server/appsmith-server/src/test/java/com/appsmith/server/services/ActionCollectionServiceTest.java:204]: Not skipping the cleanup post discussion.
Discussion:
```In the past, I have seen some test cases, where Test1 is failing a count assertion, only because Test2 created some resources. Just for that sake, I am trying to add a cleanup for applications and workspaces. This will be doing 2 things at once, (a) remove the unnecessary resources and (b) clean cache for the existing users, because deleting workspace will eventually delete the roles.```

File: app/server/appsmith-server/src/test/java/com/appsmith/server/services/ApplicationSnapshotServiceTest.java (hunk 3 above)
Comment (abhvsn) [reply app/server/appsmith-server/src/test/java/com/appsmith/server/services/ApplicationSnapshotServiceTest.java:61]: @nsarupr do we need this to be a de-facto standard for other test classes as well, considering devs may add the testcases which does not require user context just like the ones we are seeing here?

**Review by abhvsn — COMMENTED** [review]
Very minor comments

**Review by abhvsn — APPROVED** [review]

**Review by trishaanand — APPROVED** [review]
